modules/user_accounts/jsx/permissions.js

```diff
@@ -0,0 +1,337 @@
+        onMouseOut={this.handleMouseOut}>
+        <label style={textStyle}>
+          <input
+            name={"roleID[" + this.props.id + "]"}
+            type="checkbox"
+            value={this.props.id}
+            id={this.props.id}
+            checked={this.props.checked}
+            onChange={this.handleCheck}
+            disabled={this.props.disabled}
+          />
+          {this.props.name}
+        </label>
+      </div>
+    );
+  }
+});
+
+var RoleList = React.createClass({
+  render: function() {
+    var roleNodes = this.props.data.map(function(role) {
+      return (
+        <Role
+          key={role.id}
+          name={role.name}
+          id={role.id}
+          checked={role.checked}
+          disabled={role.disabled}
+          onChange={this.props.onChange}
+          onMouseOver={this.props.onMouseOver}
+          onMouseOut={this.props.onMouseOut} />
+      );
+    }, this);
+    return (
+      <div className="roleList">
+        {roleNodes}
+      </div>
+    );
+  }
+});
+
+var Permission = React.createClass({
+  handleCheck: function(e) {
+    this.props.onChange(this.props.id);
+  },
+  render: function() {
+    var textStyle = {};
+    if (this.props.disabled) {
+      textStyle = {
+        color: '#ccc',
+        cursor: 'not-allowed'
+      };
+    }
+    if (this.props.highlight) {
+      textStyle = {
+        color: 'lightblue'
+      };
+    }
+    return (
+      <div className="checkbox permission">
+        <label style={textStyle}>
+          <input
+            name={"permID[" + this.props.id + "]"}
+            type="checkbox"
+            value={this.props.id}
+            checked={this.props.checked}
+            onChange={this.handleCheck}
+            disabled={this.props.disabled}
+          />
+          {this.props.name}
+        </label>
+      </div>
+    );
+  }
+});
+
+var PermissionList = React.createClass({
+  render: function() {
+    var half = Math.ceil(this.props.data.length / 2);
+    var firstHalf = this.props.data.slice(0, half);
+    var secondHalf = this.props.data.slice(half, this.props.data.length);
+    var permissionNodesCol1 = firstHalf.map(function(permission) {
+      return (
+        <Permission
+          key={permission.id}
+          id={permission.id}
+          name={permission.name}
+          highlight={permission.highlight}
+          checked={permission.checked}
+          onChange={this.props.onChange}
+          disabled={permission.disabled} />
+      );
+    }, this);
+    var permissionNodesCol2 = secondHalf.map(function(permission) {
+      return (
+        <Permission
+          key={permission.id}
+          id={permission.id}
+          name={permission.name}
+          highlight={permission.highlight}
+          checked={permission.checked}
+          onChange={this.props.onChange}
+          disabled={permission.disabled} />
+      );
+    }, this);
+    return (
+      <div className="permissionList">
```

Comment:
To have proper spacing, divs with `col-*` classes should always be nested under a row class.